system/jlib/jsocket.cpp

```diff
@@ -1585,15 +1585,15 @@ int CSocket::wait_read(unsigned timeout)
                 ret = -1;
             }
 #else
-            if (fds[0].revents & (POLLIN | POLLHUP))
+            if ( (fds[0].revents & (POLLERR | POLLNVAL)) || (!(fds[0].revents & (POLLIN | POLLHUP))) )
             {
```

Comment:
Can error codes and 'ok' codes be present at the same time?

If not, I guess the order change doesn't matter.
If they can be.. I'm not sure what it means? Is it ok to ignore POLLIN/POLLHUP if POLLERR or POLLNVAL are present?